### PR TITLE
docs: improve compile from source docs

### LINKED ISSUE #10629: can not build ui target (make ui) on Linux
step:
```
$ git clone https://github.com/hashicorp/consul.git
$ cd consul
$ git checkout v1.8.14
$ export XC_OS=linux XC_ARCH=amd64
$ make static-assets-docker
Building Golang build container
sha256:266a702eeda31911c04861b120285d775694e1b9298817632466ea728dd3c931
==> Building Static Assets
Creating the Go Build Container with image: consul-build-go
Copying the sources from '/root/go/src/github.com/hashicorp/consul/(pkg/web_ui|GNUmakefile)' to /consul/pkg
tar: pkg/web_ui: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
Running build in container
fatal: not a git repository (or any of the parent directories): .git
bindata: Failed to stat input path 'pkg/web_ui': lstat pkg/web_ui: no such file or directory
Error: go-bindata:  exit status 1
make: *** [GNUmakefile:309: static-assets] Error 1
make: *** [static-assets-docker] Error 1

```

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
Closes #10629, in which a practitioner had trouble figuring out how to build `consul` from source.

**Changes:**
- docs: link from makefile to compile instructions
  - Some practitioners look to the makefile directly rather than to the consul website for information on how to compile from source. Link to the website instructions directly from the makefile so the practitioner can accomplish their task successfully without a careful read of the makefile.

- docs: show how to cross-compile from source

**Installation Page Screenshot**

The instructions for compiling on the current system are the default tab. This is what the cross-compile tab looks like. The red line shows an indentation issue that can't be fixed without changing the pre-commit linter rules (see comment added in PR diff).

image

### CODE REVIEW

File: website/content/docs/install/index.mdx
Comment (jkirschner-hashicorp) [inline website/content/docs/install/index.mdx:62]: Not sure there's a better way around the problem described here

**Review by dnephin — COMMENTED** [review]
Thanks for writing this up!

It makes me a little sad that we are hiding the standard `go` toolchain here.  I personally never use the makefile. If I need to build Consul I run `go build`.  If I want to cross compile I can run `GOOS=darwin GOARCH=arm go build`.

The last time I looked, the makefile goes through a bunch of bash, but it doesn't actually change anything about the `go build` command. We don't really need any special flags, and we don't have any pre-compile steps to run.

I think there are advantages to being able to use the Go toolchain directly. It makes the entire process of building and testing Consul feel less magical, and it signals to contributors that they can expect to use the Go toolchain the way they would on any other project.

I would love it if we documented just using `go build`, but if others feel there is still advantages to the makefile, I don't want to block merging this. We can discuss further in another forum.

**Comment by jkirschner-hashicorp** [comment]
@dnephin : looks like we now need a different reviewer to look at your build system changes :) so I added the Core team as a reviewer.

File: website/content/docs/install/index.mdx
Comment (jkirschner-hashicorp) [inline website/content/docs/install/index.mdx:41]: @dnephin : what does "configured properly" mean (beyond installing it)?

Comment (dnephin) [reply website/content/docs/install/index.mdx:41]: Good question, that text was there before I edited.

I suspect it means "$PATH should include the path to the git binary", or "the git binary must be in $PATH".

Comment (kisunji) [reply website/content/docs/install/index.mdx:41]: I think it's clearer to remove "and configured properly" since it ambiguously implies a "proper" configuration (we're fine with any kind of defaults from go installation).

File: website/content/docs/install/index.mdx
Comment (kisunji) [inline website/content/docs/install/index.mdx:71]: "another system" or "other systems"?

File: GNUmakefile
Comment (kisunji) [inline GNUmakefile:142]: @dnephin is `tools` no longer required?

Comment (dnephin) [reply GNUmakefile:142]: I believe that's correct. We shouldn't require any additional tools to build the binary. `tools` installs stuff for updating protos , re-generating the bindata (which you are removing), run lint, or generate mocks, but nothing that should be required to build the binary.

**Comment by jkirschner-hashicorp** [comment]
@kisunji : addressed your two minor points of feedback. Ready for re-review.

**Comment by kisunji** [comment]
> @kisunji : addressed your two minor points of feedback. Ready for re-review.

Could you try rebasing? I updated the CircleCI images and found that they had more restrictive permissions so I _think_ some stages will fail in this PR

File: website/content/docs/install/index.mdx
Comment (blake) [reply website/content/docs/install/index.mdx:53]: I've run into this same problem with the pre-commit hook. `git commit --no-verify` will allow you to commit the content unmodified.

Comment (jkirschner-hashicorp) [reply website/content/docs/install/index.mdx:53]: Thanks Blake. Didn't know that was an option!

**Comment by jkirschner-hashicorp** [comment]
@kisunji : just rebased (and addressed the tabbing issue based on @blake 's suggestion). Can you:
- Check whether the CI build works as you'd expect
- Authorize the Vercel deplyoment? (I can't do it for some reason)

File: .circleci/config.yml
Comment (kisunji) [inline .circleci/config.yml:446]:
```suggestion
            mkdir -p /home/circleci/go/bin
            cp ./bin/consul /home/circleci/go/bin/consul
```
```suggestion
            mkdir -p /go/bin
            cp ./bin/consul /go/bin/consul
```

Comment (jkirschner-hashicorp) [reply .circleci/config.yml]: Change made and branch updated.

**Review by kisunji — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/574457.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 1ff27b0b16956431d591309487d5003c1ca02bb3 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 1ff27b0b16956431d591309487d5003c1ca02bb3 onto `release/1.11.x` succeeded!

**Pull request merged**
